### ISSUE TITLE
imp DeprecationWarning

### ISSUE THREAD

**Issue by dpoznik** [Netflix/metaflow#562]
I get the following warning from `pytest`:
```
==================================== warnings summary =====================================
../../../../usr/local/lib/python3.8/site-packages/metaflow/R.py:2
  /usr/local/lib/python3.8/site-packages/metaflow/R.py:2: 
DeprecationWarning: the imp module is deprecated in favour of importlib; 
see the module's documentation for alternative uses
    import imp

-- Docs: https://docs.pytest.org/en/stable/warnings.html
```
I think this change would address the warning, but only for `py3`:
https://github.com/Netflix/metaflow/compare/master...dpoznik:replace-imp

I can, of course, just silence the warning on my end if there's not a good fix.

**Comment by savingoyal** [comment]
Thanks for the issue! The change would definitely address the warning, but only for py3. Flows using Python2 underneath will start breaking - maybe we can introduce a `try... except...` clause to handle both py2 and py3 properly?

**Comment by dpoznik** [comment]
> maybe we can introduce a `try... except...` clause to handle both py2 and py3 properly?

OK, sounds good. I'll put in a PR in a moment...

**Comment by dpoznik** [comment]
Resolved in #1396
